VDA-DID-Linkage/contracts/VeridaDIDLinkage.sol

```diff
@@ -0,0 +1,256 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+// import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+// import "@openzeppelin/contracts-upgradeable/utils/structs/EnumerableSetUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/StringsUpgradeable.sol";
+
+import "@verida/vda-verification-contract/contracts/VDAVerificationContract.sol";
+
+import "./IVeridaDIDLinkage.sol";
+import "./StringUtils.sol";
+import "./EnumerableSet.sol";
+
+contract VeridaDIDLinkage is VDAVerificationContract,
+    IVeridaDIDLinkage
+{
+
+    using EnumerableSetUpgradeable for EnumerableSetUpgradeable.AddressSet;
+    using EnumerableSet for EnumerableSet.StringSet;
+
+    /**
+     * @notice Signer type for each identifier type
+     * @dev mapping of identifier type => signer type
+     */
+    mapping(string => string) private _identifierType;
+
+    /**
+     * @notice Mapping of identifier & it's controller did
+     * @dev mapping of identifier => did. did is not an address. Ex of did : "did:vda:0x..."
+     */
+    mapping(string => string) private _identifierController;
+
+    /**
+     * @notice Mapping of did & identifiers list that linked to did
+     * @dev mapping of did => identifier set
+     */
+    mapping(string => EnumerableSet.StringSet) private _didIdentifiers;
+
+    /**
+     * @notice Initialize
+     */
+    function initialize() public initializer {
+        __VDAVerificationContract_init();
+    }
+
+    /**
+     * @dev See {IVeridaDIDLinkage}
+     */
+    function addIdentifierType(string calldata identifierTypeId, string calldata signerType) external onlyOwner {
```

Comment:
For efficiency should use a boolean for `signerType`. Where true = self-signed, false = trusted. Or a uint.

Reply:
Updated